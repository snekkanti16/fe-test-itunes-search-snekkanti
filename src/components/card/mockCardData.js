export const mockTrackData = {
    "key":"track0",
    "wrapperType": "track",
    "kind": "song",
    "artistId": 5869117,
    "collectionId": 1580652340,
    "trackId": 1580652352,
    "artistName": "Lil Wayne",
    "collectionName": "Tha Carter IV (Complete Edition)",
    "trackName": "John (feat. Rick Ross)",
    "collectionCensoredName": "Tha Carter IV (Complete Edition)",
    "trackCensoredName": "John (feat. Rick Ross)",
    "artistViewUrl": "https://music.apple.com/us/artist/lil-wayne/5869117?uo=4",
    "collectionViewUrl": "https://music.apple.com/us/album/john-feat-rick-ross/1580652340?i=1580652352&uo=4",
    "trackViewUrl": "https://music.apple.com/us/album/john-feat-rick-ross/1580652340?i=1580652352&uo=4",
    "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/99/0a/4a/990a4a34-bc4b-de8e-1b06-1ed0b47492c0/mzaf_15268783310218388431.plus.aac.p.m4a",
    "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/a9/75/07/a97507a8-4316-2b93-3deb-dff6041d2d79/source/30x30bb.jpg",
    "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/a9/75/07/a97507a8-4316-2b93-3deb-dff6041d2d79/source/60x60bb.jpg",
    "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/a9/75/07/a97507a8-4316-2b93-3deb-dff6041d2d79/source/100x100bb.jpg",
    "collectionPrice": 12.99,
    "trackPrice": 1.29,
    "releaseDate": "2011-03-24T12:00:00Z",
    "collectionExplicitness": "explicit",
    "trackExplicitness": "explicit",
    "discCount": 1,
    "discNumber": 1,
    "trackCount": 21,
    "trackNumber": 9,
    "trackTimeMillis": 286973,
    "country": "USA",
    "currency": "USD",
    "primaryGenreName": "Hip-Hop/Rap",
    "contentAdvisoryRating": "Explicit",
    "isStreamable": true
};

export const mockCollectionData = {
    "key":"collection1",
    "wrapperType": "collection",
    "collectionType": "Album",
    "artistId": 180586,
    "collectionId": 273750173,
    "amgArtistId": 4427,
    "artistName": "Daryl Hall & John Oates",
    "collectionName": "The Very Best of Daryl Hall & John Oates",
    "collectionCensoredName": "The Very Best of Daryl Hall & John Oates",
    "artistViewUrl": "https://music.apple.com/us/artist/daryl-hall-john-oates/180586?uo=4",
    "collectionViewUrl": "https://music.apple.com/us/album/the-very-best-of-daryl-hall-john-oates/273750173?uo=4",
    "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/21/48/21/214821c2-c2e3-bd4b-3d59-94fd3430da13/source/60x60bb.jpg",
    "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/21/48/21/214821c2-c2e3-bd4b-3d59-94fd3430da13/source/100x100bb.jpg",
    "collectionPrice": 10.99,
    "collectionExplicitness": "notExplicit",
    "trackCount": 18,
    "copyright": "â„— This compilation (P) 2001 Sony Music Entertainment",
    "country": "USA",
    "currency": "USD",
    "releaseDate": "2001-01-23T08:00:00Z",
    "primaryGenreName": "Rock"
};
