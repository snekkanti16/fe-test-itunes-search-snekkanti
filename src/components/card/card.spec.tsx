import React from "react";
import { render } from "@testing-library/react";
import Card from './index';
import {mockTrackData, mockCollectionData} from "./mockCardData";

describe("/", () => {
    it("should render the Track Card", () => {
      const { container } = render(<Card {...mockTrackData} />);
      expect(container).toMatchSnapshot();
    });
    it("should render the Album Card", async () => {
        const { container } = render(<Card {...mockCollectionData} />);
        expect(container).toMatchSnapshot();
    });
  });