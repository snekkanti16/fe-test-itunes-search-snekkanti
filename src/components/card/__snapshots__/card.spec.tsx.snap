// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ should render the Album Card 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-8"
    data-testid="grid"
  >
    <div
      class="MuiPaper-root makeStyles-paper-9 makeStyles-cardBackground1-13 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <img
        alt="collection"
        class="makeStyles-card-10"
        src="https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/21/48/21/214821c2-c2e3-bd4b-3d59-94fd3430da13/source/100x100bb.jpg"
      />
      <div
        class="makeStyles-cardType-12"
      >
        (
        Album
        )
      </div>
      <div
        class="makeStyles-desc-11"
      >
        Artist Name : 
        Daryl Hall & John Oates
      </div>
      <div
        class="makeStyles-desc-11"
      >
        Collection Name : 
        The Very Best of Daryl Hall & John Oates
      </div>
    </div>
  </div>
</div>
`;

exports[`/ should render the Track Card 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1"
    data-testid="grid"
  >
    <div
      class="MuiPaper-root makeStyles-paper-2 makeStyles-cardBackground2-7 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <img
        alt="track"
        class="makeStyles-card-3"
        src="https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/a9/75/07/a97507a8-4316-2b93-3deb-dff6041d2d79/source/100x100bb.jpg"
      />
      <div
        class="makeStyles-cardType-5"
      >
        (
        song
        )
      </div>
      <div
        class="makeStyles-desc-4"
      >
        Artist Name : 
        Lil Wayne
      </div>
      <div
        class="makeStyles-desc-4"
      >
        Track Name : 
        John (feat. Rick Ross)
      </div>
      <div
        class="makeStyles-desc-4"
      >
        Collection Name : 
        Tha Carter IV (Complete Edition)
      </div>
    </div>
  </div>
</div>
`;
